khal/controllers.py

```diff
@@ -458,20 +476,14 @@ def edit_event(event, collection, locale, allow_quit=False, width=80):
             except:  # noqa
                 echo("error parsing range")
         elif choice == "repeat":
-            recur = event.recurobject
-            freq = recur["freq"] if "freq" in recur else ""
-            until = recur["until"] if "until" in recur else ""
-            if not freq:
-                freq = 'None'
-            freq = prompt('frequency (or "None")', freq)
-            if freq == 'None':
-                event.update_rrule(None)
+            freq = prompt_none("frequency", event.recurobject.get("freq"), lower=True)
+            if freq:
+                until = prompt_none("until", event.recurobject.get("until"))
+                interval = prompt_none("interval", event.recurobject.get("interval"), int)
```

Comment:
I also notice you've changed from using the dict syntax to the `.get()` method. As you're not specifying a default I don't see any benefit to this.

Reply:
The dict syntax throws a KeyError exception on a missing key while get() returns the default value (None).  Before the code did:

recur = event.recurobject
freq = recur["freq"] if "freq" in recur else ""
if not freq:
  freq = 'None'
freq = prompt('frequency (or "None")', freq)

I thought it was clearer leaving it as None, and handle the magic "None" value in the prompt_none helper as it's really work-around for prompt conflating default values and optional/required values.